modules/client-handler/src/main/java/org/apache/ignite/client/handler/requests/compute/ClientComputeExecuteRequest.java

```diff
@@ -66,7 +69,14 @@ public static CompletableFuture<Void> process(
      * @param in Unpacker.
      * @return Args array.
      */
-    static Object[] unpackArgs(ClientMessageUnpacker in) {
-        return in.unpackObjectArrayFromBinaryTuple();
+    static Object[] unpackArgs(ClientMessageUnpacker in, String jobClassName) {
```

Comment:
Ideally we should not unpack arguments if we are not on the target node at the moment.